### PR TITLE
Makefile: refactoring .PHONY

### LINKED ISSUE #52259: make error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

When there is a Docker file in the root path of the project, executing the make docker command will result in an error
image

### PR DESCRIPTION
**Pull request by imalasong**
### What problem does this PR solve?

PHONY is set on one line, which makes it cumbersome and difficult to maintain as more functions are added. And I found that many functions were not set to enter PHONY

Now set your own PHONY for each function

The changes are as follows:

    1、delete tazel (not used)
    2、modify: These are all set on the top line of the corresponding function：help all clean test server dev benchkv benchraw check checklist parser tidy ddltest build_br build_lightning build_lightning-ctl build_dumpling ut bazel_build bazel_prepare bazel_test check-file-perm check-bazel-prepare bazel_lint precheck
    3、addition: docker、buildsucc、linux...etc

Issue Number: close #52259

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -352,12 +424,15 @@ build_for_br_integration_test:
 	) || (make failpoint-disable && exit 1)
 	@make failpoint-disable
 
+.PHONY:br_unit_test
```
Comment (lance6716) [inline Makefile:427]:
```suggestion
.PHONY: br_unit_test
```

**Review by lance6716 — APPROVED** [review]

**Comment by imalasong** [comment]
Do I need to resolve this conflict? Merge the master to my PR branch first, resolve it before submitting

**Comment by hawkingrei** [comment]
> Do I need to resolve this conflict? Merge the master to my PR branch first, resolve it before submitting

Yes. Please solve this problem.

**Review by hawkingrei — APPROVED** [review]

**Pull request closed without merging**